pkg/controller/storagecluster/update.go

```diff
@@ -233,28 +262,28 @@ func storkUnschedulableAnnotationHelper(k8sClient client.Client, nodeName string
 		return fmt.Errorf("failed to get node %s: %w", nodeName, err)
 	}
 	needsUpdate := false
-	val, ok := node.Annotations[constants.StorkAnnotationUnschedulable]
-	if unschedulable && val != "true" {
+	val := nodeMarkedUnschedulable(node)
+	if unschedulable && !val {
 		if node.Annotations == nil {
 			node.Annotations = make(map[string]string)
 		}
-		node.Annotations[constants.StorkAnnotationUnschedulable] = "true"
+		node.Annotations[constants.AnnotationUnschedulable] = "true"
 		needsUpdate = true
-	} else if !unschedulable && ok {
-		delete(node.Annotations, constants.StorkAnnotationUnschedulable)
+	} else if !unschedulable && val {
+		delete(node.Annotations, constants.AnnotationUnschedulable)
 		needsUpdate = true
 	}
 	if !needsUpdate {
 		return nil
 	}
 	if unschedulable {
-		logrus.Infof("Adding annotation %s to node %s", constants.StorkAnnotationUnschedulable, nodeName)
+		logrus.Infof("Adding annotation %s to node %s", constants.AnnotationUnschedulable, nodeName)
 	} else {
-		logrus.Infof("Removing annotation %s from node %s", constants.StorkAnnotationUnschedulable, nodeName)
+		logrus.Infof("Removing annotation %s from node %s", constants.AnnotationUnschedulable, nodeName)
```

Comment:
nit..We can consider moving these print statements inside the above if/else blocks. 
Logically it's correct since we are updating needsUpdate inside the if/else and returning if do no need update.
But from the readability standpoint, it's a little confusing as it suggests that we are printing "Adding annotation" if unschedulable is true(and the only condition) and vice versa which is not true.